### PR TITLE
Exponential Backoff for ByteFIFO

### PR DESCRIPTION
**Pull request by zeripath**
This PR is another in the vein of queue improvements. It suggests an
exponential backoff for bytefifo queues to reduce the load from queue
polling. This will mostly be useful for redis queues.

Related #15542

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by zeripath** [comment]
Related #15542

**Comment by zeripath** [comment]
This could easily be moved in to #15693

**Review by silverwind — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Review by lafriks — DISMISSED** [review]
small nit but otherwise looks good

File: modules/queue/queue_bytefifo.go [hunk 1]
```diff
@@ -126,21 +142,23 @@ func (q *ByteFIFOQueue) readToChan() {
 			if err != nil {
```
Comment (lunny) [inline modules/queue/queue_bytefifo.go]: Just change `default` to a timer.C, after the time is out, try to Pop . This will make the code simpler.

Comment (zeripath) [reply modules/queue/queue_bytefifo.go]: There is no backoff when there is a successful pop

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
->  #15793

**Pull request merged**